server/utils/appInsightsUtils.ts

```diff
@@ -1,13 +1,43 @@
 import type { TelemetryClient } from 'applicationinsights'
-import { DistributedTracingModes, defaultClient, setup } from 'applicationinsights'
+import { Contracts, DistributedTracingModes, defaultClient, setup } from 'applicationinsights'
+import type { EnvelopeTelemetry } from 'applicationinsights/out/Declarations/Contracts'
 
 import { buildNumber, packageData } from '../applicationVersion'
+import type { User } from '../data'
+
+type ContextObjectWithUser = {
+  res?: {
+    locals?: {
+      user: Partial<User>
+    }
+  }
+}
+
+export type ContextObject = Record<string, ContextObjectWithUser> | undefined
 
 export default class AppInsightsUtils {
+  static addUserDataToRequests = (envelope: EnvelopeTelemetry, contextObjects: ContextObject): boolean => {
+    const isRequest = envelope.data.baseType === Contracts.TelemetryTypeString.Request
+    if (isRequest) {
+      const { username, activeCaseLoadId } = contextObjects?.['http.ServerRequest']?.res?.locals?.user || {}
+      if (username && envelope.data.baseData) {
```

Comment:
Could we condense this into one conditional? The only benefit of having two is not reading two properties, and it would flatten this block a bit/make it more readable